model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/entities/AuthenticatedClientSessionEntity.java

```diff
@@ -83,6 +86,17 @@ public void setTimestamp(int timestamp) {
         this.timestamp = timestamp;
     }
 
+    public int getUserSessionStarted() {
+        String started = getNotes().get(USER_SESSION_STARTED_AT_NOTE);
+        // Fallback to timestamp if "started" note is not available.
+        return started == null ? timestamp : Integer.parseInt(started);
+    }
+
+    public int getStarted() {
+        String started = getNotes().get(AuthenticatedClientSessionModel.STARTED_AT_NOTE);
+        return started == null ? timestamp : Integer.parseInt(started);
+    }
+
```

Comment:
These methods belong to either `AuthenticatedClientSessionModel` (not `AuthenticatedClientSessionEntity`) or into an utility method.

Reply:
I need to maintain both in the infinispan project. This class is not related at all to `AuthenticatedClientSessionModel` so I cannot reuse the methods from there. So creating them here is more or less the same than creating a utility class in the infinispan section.

Let me know if you have a better idea.